### DISCUSSION TITLE
What to do when Dagster fails when running in VSCode terminal with an "msizes 8/7 disagree" error

### DISCUSSION THREAD

**Discussion by gibsondan** [dagster-io/dagster#12721] [Q&A] [answered]
When launching Dagster in the VSCode terminal, an error like this appears:

python(64011,0x1f6d68140) malloc: Corruption of free object 0x14be04420: msizes 8/7 disagree
python(64011,0x1f6d68140) malloc: *** set a breakpoint in malloc_error_break to debug
python(64010,0x1f6d68140) malloc: Corruption of free object 0x15b64caf0: msizes 8/7 disagree
python(64010,0x1f6d68140) malloc: *** set a breakpoint in malloc_error_break to debug

What should you do?

**Comment by gibsondan** [comment, marked as answer]
This problem occurs to a MallocNanoZone environment variable set in VSCode on M1/M2 Macs that changes Malloc behavior. To fix the problem, set the following in VSCode settings:

```
"terminal.integrated.env.osx": {
   "MallocNanoZone": "1",
}
```

**Reply by fentresspaul61B** [reply]
Thank you. This worked for me. 
As an additional note, after I updated my VSCode settings, and re ran my dagit command, it did not make a difference initially. So I had to exit VSCode, restart it, re download dagster and dagit using this command: 
```
pip install dagster dagit --find-links=https://github.com/dagster-io/build-grpcio/wiki/Wheels
```
All of this in combination with the new setting proposed above got my dagit UI working.

**Comment by remigabillet** [comment]
I just ran into this error, and the solution proposed by @gibsondan worked right away. 🙏
